boxoffice/views/admin_discount.py

```diff
@@ -51,11 +65,35 @@ def jsonify_discount_policies(data_dict):
 @render_with({'text/html': 'index.html', 'application/json': jsonify_discount_policies}, json=True)
 def admin_discount_policies(organization):
     query = request.args.get('search')
-    if query:
-        discount_policies = DiscountPolicy.query.filter(DiscountPolicy.title.ilike('%{query}%'.format(query=query))).all()
+    page = request.args.get('page', 1)
+
+    if request.is_xhr:
+        if query:
+            discount_policies = DiscountPolicy.query.filter(DiscountPolicy.title.ilike('%{query}%'.format(query=query)))
```

Comment:
For instance, if I search for `JSFoo`, this might theoretically return 20 results, which in turn has to be paginated.